query_execution/QueryManagerBase.hpp

```diff
@@ -338,6 +285,19 @@ class QueryManagerBase {
    **/
   virtual bool checkRebuildOver(const dag_node_index index) const = 0;
 
+  /**
+   * @brief Check if the given operator has ever a normal work order.
+   *
+   * @param index The index of the given operator in the DAG.
+   *
+   * @return True if the operator has a normal work order, false otherwise.
+   **/
+  virtual bool hasEverNormalWorkOrders(const dag_node_index index) const = 0;
```

Comment:
Suggest to rename this function to ``hasEverProducedNormalWorkOrder`` to make more sense.